src/merger/file_manager.rs

```diff
@@ -47,43 +63,130 @@ impl FileManager {
-                        DataType::Boolean => Arc::new(BooleanArray::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Int8 => Arc::new(Int8Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Int16 => Arc::new(Int16Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Int32 => Arc::new(Int32Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::UInt8 => Arc::new(UInt8Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::UInt16 => Arc::new(UInt16Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::UInt32 => Arc::new(UInt32Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::UInt64 => Arc::new(UInt64Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Float16 => Arc::new(Float16Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Float32 => Arc::new(Float32Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Float64 => Arc::new(Float64Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Timestamp(_, _) => Arc::new(TimestampNanosecondArray::from(vec![None; record_batch.num_rows()])) as ArrayRef, // Assuming nanoseconds precision
-                        DataType::Date64 => Arc::new(Date64Array::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Time32(_) => Arc::new(Time32SecondArray::from(vec![None; record_batch.num_rows()])) as ArrayRef, // Assuming seconds
-                        DataType::Time64(_) => Arc::new(Time64NanosecondArray::from(vec![None; record_batch.num_rows()])) as ArrayRef, // Assuming nanoseconds
-                        DataType::Duration(_) => Arc::new(DurationNanosecondArray::from(vec![None; record_batch.num_rows()])) as ArrayRef, // Assuming nanoseconds
-                        DataType::Interval(_) => Arc::new(IntervalYearMonthArray::from(vec![None; record_batch.num_rows()])) as ArrayRef, // Assuming year-month interval
-                        DataType::Binary => Arc::new(BinaryArray::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::FixedSizeBinary(_) => Arc::new(FixedSizeBinaryArray::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::LargeBinary => Arc::new(LargeBinaryArray::from(vec![None; record_batch.num_rows()])) as ArrayRef,
-                        DataType::Utf8 => Arc::new(StringArray::from(vec![String::from(""); record_batch.num_rows()])) as ArrayRef,
+                        DataType::Int64 => {
+                            Arc::new(Int64Array::from(vec![None; record_batch.num_rows()]))
```

Comment:
this can probably be done in a binding: `let rows = vec![None; record_batch.num_rows()]`